### PR TITLE
Drop Apache 2.2 support

### PR DESCRIPTION
**Pull request by ekohl**
This is likely incomplete and untested at the moment. However, it shows how big the cleanup is.

### CODE REVIEW

File: manifests/default_mods.pp
Comment (ekohl) [inline manifests/default_mods.pp:21]: Here I wonder what the best approach is. Why doesn't Amazon Linux support the `use_systemd` flag? And I think Amazon Linux 2022 is also out, which also has systemd. Should we simplify? Related: in other places I dropped Amazon Linux 1 support since it's based on EL6.

**Comment by ekohl** [comment]
This is now green.

**Comment by david22swan** [comment]
@ekohl This look's good to me so far, haven't finished the review yet though.
Will need to wait a bit on the merge though, give some time to get any other PRs that are near completion in a merged so we can cut a pre-major release, followed by some work to get any other backwards incompatible changes that we want in.

**Comment by david22swan** [comment]
@ekohl As a not, there was previously work done by a former member of our team to remove 2.2 support from the module. While it was unable to be merged at the time due to push back from customers and the community I have made sure to keep the branch saved.
It is currently pushed up as 2.2removal on my fork. Was planning to work on rebasing it when I had the free time but if you want to use it as a help with this PR here it may be off assistance.

**Comment by ekohl** [comment]
I'm pretty occupied for the next week or 2 so feel free to use my code.

**Comment by david22swan** [comment]
@ekohl From how you had responded had thought that this needed more work to get it over the line, but looking now it seems to be complete to me. In your opinion is there anything that needs one on it other than rebasing?

**Comment by ekohl** [comment]
Since https://github.com/puppetlabs/puppetlabs-apache/pull/2329#issuecomment-1289803115 I think this is ready. I've just rebased this.

**Comment by david22swan** [comment]
Ok, will give it one last review and make a clean release before merging it in.

Apologies for my misunderstanding of the situation

**Comment by ekohl** [comment]
No worries, next time I'll be more explicit. I do use the draft feature quite explicitly to mark my work as in progress.

**Comment by ekohl** [comment]
No, it's because I only partially addressed #2331. The Puppet code gave a merge conflict so I addressed that and fixed up the template as well, but forgot the tests.

File: manifests/version.pp
Comment (david22swan) [inline manifests/version.pp:9]: Seeing as how everything has been removed should this not ust be deleted?

Comment (ekohl) [reply manifests/version.pp:9]: I wonder. It does still have class parameters that can be used. Those could also be added to `params.pp`, but I don't know what's best here.

Comment (david22swan) [reply manifests/version.pp:9]: It's a hard question. I guess for now it may be best to leave them in place, at least until hiera is properly implemented within the module. Just so we have a clear place to set them.

**Review by david22swan — APPROVED** [review]
LGTM
Will merge once I can get a clean pre-major release out

**Comment by chelnak** [comment]
This is awesome ❤️

**Comment by david22swan** [comment]
Well, that's that
Thank you for all the work :)

**Pull request merged**
